aesm-client/src/windows/win_aesm_client.rs

```diff
@@ -0,0 +1,393 @@
+    Data1: 0x50AFD900,
+    Data2: 0xF309,
+    Data3: 0x4557,
+    Data4: [0x8F, 0xCB, 0x10, 0xCF, 0xAB, 0x80, 0x2C, 0xDD],
+};
+
+trait HresultExt:Sized {
+    fn into_io_error(self) -> IoResult<Self>;
+}
+
+impl HresultExt for HRESULT {
+    fn into_io_error(self) -> IoResult<HRESULT> {
+        if SUCCEEDED(self) {
+            Ok(self)
+        } else {
+            Err(IoError::from_raw_os_error(self))
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct AesmClient {
+    interface: *mut AesmInterface,
+}
+
+impl Drop for AesmClient {
+    fn drop(&mut self) {
+        unsafe {
+            if let Some(release) = (*(*self.interface).vtbl).release {
+                release(self.interface);
+            }
+            CoUninitialize();
+        }
+    }
+}
+
+impl AesmClient {
+    pub fn new() -> Result<Self> {
+        AesmClient::create_instance()
+    }
+
+    fn initialize_com(&self) -> Result<HRESULT> {
+        unsafe {
+            CoInitializeEx(
+                ptr::null_mut(),
+                COINIT_MULTITHREADED | COINIT_DISABLE_OLE1DDE,
+            )
+                .into_io_error()
+                .map_err(|e| Error::AesmCommunication(IoError::new(ErrorKind::Other, format!("Failed to initialize COM: {}", e))) )
+        }
+    }
+
+    pub fn try_connect(&self) -> Result<()> {
+        self.initialize_com().map(|_| ())
+    }
+
+    fn create_instance() -> Result<Self> {
+        let mut instance: *mut AesmInterface = std::ptr::null_mut();
+        unsafe {
+            CoInitializeEx(
+                ptr::null_mut(),
+                COINIT_MULTITHREADED | COINIT_DISABLE_OLE1DDE,
+            )
+                .into_io_error()
+                .map_err(|e| IoError::new(ErrorKind::Other, format!("Failed to initialize COM: {}", e)) )?;
+
+            CoCreateInstance(
+                &CLSID_AESMINTERFACE,
+                ptr::null_mut(),
+                CLSCTX_ALL,
+                &IID_IAESMINTERFACE,
+                &mut instance as *mut _ as *mut *mut c_void,
+            )
+                .into_io_error()
+                .map_err(|e|
+                    IoError::new(ErrorKind::Other, format!("Fail to create Aesm Interface {}", e)) )?;
+        }
+        Ok(AesmClient {
+            interface: instance,
+        })
+    }
+
+    pub fn init_quote(&self) -> Result<QuoteInfo> {
+        let mut target_info: Vec<u8> = vec![0; sgx_isa::Targetinfo::UNPADDED_SIZE];
+        let mut gid: Vec<u8> = vec![0; 4];
+        let mut error: AesmError = 0;
+        unsafe {
+            if let Some(init_quote) = (*(*self.interface).vtbl).init_quote {
+                init_quote(
+                    self.interface,
+                    target_info.as_mut_ptr(),
+                    target_info.len() as _,
+                    gid.as_mut_ptr(),
+                    gid.len() as _,
+                    &mut error as _,
+                ).into_io_error().
+                    map_err(|e| IoError::new(ErrorKind::Other,
```

Comment:
No need to add an error message if the function is only doing a single operation. Same thing for other functions below.